### ISSUE TITLE
When a selected expression begins/ends with a bracket, the "Extract method/variable" command misbehaves

### ISSUE THREAD

**Issue by YtotheWT** [microsoft/pylance-release#1768]
## Environment data

-   Language Server version: 2021.9.0
-   OS and version: Windows 10 20H2
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.6

## Expected behaviour

Not to leave the leading/trailing bracket where it is originally

## Actual behaviour

Pylance leaves the leading/trailing bracket where it is originally, which may cause errors

## Logs

When I select `(a + b) // 2`:
```
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
```

When I click "extract variable":
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: c:\Users\Admin\ywt\2.py ...
[BG(1)]   parsing: c:\Users\Admin\ywt\2.py (1ms)
[BG(1)]   binding: c:\Users\Admin\ywt\2.py (0ms)
[BG(1)]   checking: c:\Users\Admin\ywt\2.py (4ms)
[BG(1)] analyzing: c:\Users\Admin\ywt\2.py (6ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: resumeAnalysis
Background analysis message: getDiagnosticsForRange
[FG] parsing: c:\Users\Admin\ywt\2.py (0ms)
[FG] binding: c:\Users\Admin\ywt\2.py (0ms)
Background analysis message: getSemanticTokens delta
[BG(1)] getSemanticTokens delta previousResultId:1630760446008 at c:\Users\Admin\ywt\2.py (0ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: markFilesDirty
Background analysis message: analyze
```

## Code Snippet / Additional information

Original code:
```python
a = 1
b = 2
print((a + b) // 2, end='')
```
When I select `(a + b) // 2`, click the light bulb, choose "Extract variable", press Enter, this is what I see:
```python
a = 1
b = 2
new_var = a + b) // 2
print((new_var, end='')
```
As you can see, the `(` following `print` was still in Line 4 where it was originally, instead of at the right hand side of `=` in Line 3.
![屏幕截图 2021-09-04 210014](https://user-images.githubusercontent.com/62426149/132095574-3a4e0454-0c08-424b-b33c-d82b5566b010.png)
![屏幕截图 2021-09-04 211150](https://user-images.githubusercontent.com/62426149/132095578-96e8edab-0eae-4ec0-b350-763ec39924f2.png)

And this error also happens on "Extract method" command.

**Comment by debonte** [comment]
@CodeCrazy-ywt, are you still able to repro this issue? I'm seeing something different, but I'm guessing it's related:

Using your example above, if I select "(a + b) // 2", I don't see the code action lightbulb, and if I press Ctrl+. I'm told that there are no available code actions:

![image](https://user-images.githubusercontent.com/12820898/150896490-3396fcd4-4bf1-4246-8df7-92f7ffcb9e88.png)

But if I limit my selection to "a + b) // 2" (i.e. don't include the open paren), the Extract Method and Extract Variable code actions are available, but result in the invalid Python syntax that you described above:

![image](https://user-images.githubusercontent.com/12820898/150896726-875b159e-d8a2-4533-b04d-320fc1043151.png)

Is this what you're seeing (now) too?

**Comment by CodeCrazy-ywt** [comment]
Sorry for late response. What I see is exactly the same as yours.

**Comment by debonte** [comment]
FYI, this also causes _getIndentation to misbehave. Getting the indentation for the line after `return x == 0` in the following code results in a single dedent as expected:

```python
def func(x):
    if x != None:
        return x == 0
```

Whereas, getting the indentation for the line after `return (x == 0)` results in zero indentation:

```python
def func(x):
    if x != None:
        return (x == 0)
```

This happens because calling `findNodeByOffset` with the end offset of the `CloseParenthesis` token returns the `Module` ParseNode instead of the `BinaryOperation`.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221121-2-November-2022-PreRelease)
